va/va.go

```diff
@@ -488,6 +488,14 @@ func (va VAImpl) fetchHTTP(identifier string, token string) ([]byte, string, *ac
 		return nil, url.String(), acme.MalformedProblem(
 			fmt.Sprintf("Invalid URL %q\n", url.String()))
 	}
+	// trim the :80 port of the url off the host header
```

Comment:
Let's move this above where we create the URL; we should create a variable `host` that contains either the host:port or host alone, depending on the port. Then we can stick that directly into the `url.URL` when we create it.

Reply:
This comment was from an earlier draft of feedback, sorry I forgot to delete it. Let's resolve the spec issue before we move forward with this edit.